docs/style-guide.md

```diff
@@ -85,27 +115,91 @@ There are three areas which may warrant a technically public API that we don't c
+This is a major difference for users coming from the C++ family of languages.
+
+Since mutability is controlled via ownership and borrow checker rules, if the getter is just returning a reference, and
+the setter is only assigning the value, then exposing a public field is recommended for ergonomics.
+
+#### Example
+
+```rust
+enum DateTimeStyle {
+    Long,
+    Short,
+}
+
+struct DateTimeOptions {
+    pub time_style: DateTimeStyle,
+    pub date_style: DateTimeStyle,
+}
+
+fn main() {
+    let mut opts = DateTimeOptions {
+        time_style: DateTimeStyle::Short,
+        date_style: DateTimeStyle::Long,
+    }
+  
+    // Override
+    opts.time_style = DateTimeStyle::Long;
+}
+```
+
+Since the `date_style` and `time_style` fields can accept only valid values, and the getter/setter methods
+would not perform any additional operations, those fields can be exposed as public.
+
+In all other situations, where the getter/setter are fallible, perform additional computations or optimizations, or an invariant between the fields has to be achieved, it is advised to use an equivalent getter/setter model:
 
-Unless there is a compelling use case, it seems very sensible to never use public fields in structs. This allows for clean encapsulation with essentially no downside for ICU4X. This post sums things up nicely, which I am summarizing below:
+```rust
+enum DateTimeStyle {
+    Long,
+    Short,
+}
+
+struct DateTimeOptions {
+    meta: Option<TinyStr8>,
+}
 
-For a **pub** field `bar` in a **pub** struct `Foo`:
+impl DateTimeOptions {
+    pub fn get_time_style(&self) -> Option<&DateTimeStyle>;
+    pub fn set_time_style(&mut self, input: DateTimeStyle) -> Result;
 
-* Construction: users can create a `Foo { bar: 42 }`.
-  * This also means it’s a breaking change for you to add or remove any fields (and using a Default trait won't help you here since you still have to rely on every caller using it to avoid adding a new field being a breaking change).
-* Mutation: if the user has some `mut var: Foo` or `&mut Foo`, they can change the value without notice.
-  * While this won't cause any race conditions in Rust, it does prevent a whole class of useful optimizations for types around internal consistency (e.g. you cannot cache otherwise expensive hash-code values).
-* Partial borrow: if a user binds `let var = &foo.bar;` they can still use other fields without any complaint from the borrow checker.
+    pub fn get_date_style(&self) -> Option<&DateTimeStyle>;
+    pub fn set_date_style(&mut self, input: DateTimeStyle) -> Result;
+    
+    pub fn get_meta(&self) -> Option<&str>;
+    pub fn set_meta(&mut self, input: &str) -> Result;
+}
 
-You can always supply public inlinable getters to access fields, and non-public fields are still directly accessible by the current module and any sub-modules.
+fn main() {
+    let mut opts = DateTimeOptions {
+        time_style: DateTimeStyle::Short,
```

Comment:
This doesn't make sense. `time_style` and `date_style` are not fields of `DateTimeOptions`, and `meta` is private.